code/modules/antagonists/vampire/vampire_powers/dantalion_powers.dm

```diff
@@ -0,0 +1,203 @@
+		return FALSE
+
+	var/enthrall_safe = FALSE
+	for(var/obj/item/implant/mindshield/L in C)
+		if(L.implanted)
+			enthrall_safe = TRUE
+			break
+	var/datum/antagonist/vampire/V = user.mind.has_antag_datum(/datum/antagonist/vampire)
+	if(V.subclass.thrall_cap <= length(user.mind.som.serv))
+		to_chat(user, "<span class='warning'>You don't have enough power to enthrall any more people!</span>")
+		return FALSE
+	if(enthrall_safe || C.mind.has_antag_datum(/datum/antagonist/vampire) || C.mind.has_antag_datum(/datum/antagonist/mindslave))
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>You feel a familiar sensation in your skull that quickly dissipates.</span>")
+		return FALSE
+	if(C.mind.isholy)
+		C.visible_message("<span class='warning'>[C] seems to resist the takeover!</span>", "<span class='notice'>Your faith in [SSticker.Bible_deity_name] has kept your mind clear of all evil.</span>")
+		return FALSE
+	return TRUE
+
+/obj/effect/proc_holder/spell/vampire/enthrall/proc/handle_enthrall(mob/living/user, mob/living/carbon/human/H)
+	if(!istype(H))
+		return FALSE
+
+	var/greet_text = "<b>You have been Enthralled by [user.real_name]. Follow [user.p_their()] every command.</b>"
+	H.mind.add_antag_datum(new /datum/antagonist/mindslave/thrall(user.mind, greet_text))
+	H.Stun(2)
+	user.create_log(CONVERSION_LOG, "vampire enthralled [H.real_name]")
+	H.create_log(CONVERSION_LOG, "was vampire entrhalled by [user.real_name]")
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune
+	name = "Commune"
+	desc = "Talk to your thralls telepathically."
+	gain_desc = "You have gained the ability to commune with your thralls."
+	action_icon_state = "vamp_communication"
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/create_new_handler() //so thralls can use it
+	return
+
+/datum/spell_targeting/select_vampire_network/choose_targets(mob/user, obj/effect/proc_holder/spell/spell, params, atom/clicked_atom) // Returns the vampire and their thralls. If user is a thrall then it will look up their master's network
+	var/list/mob/living/targets = list()
+	var/datum/antagonist/vampire/V
+	V = user.mind.has_antag_datum(/datum/antagonist/vampire) // if the user is a vampire
+
+	if(!V)
+		for(var/datum/mind/M in user.mind.som.masters) // if the user is a thrall
+			V = M.has_antag_datum(/datum/antagonist/vampire)
+			if(V)
+				break
+
+	if(!V)
+		return
+
+	for(var/datum/mind/thrall in V.owner.som.serv)
+		targets += thrall.current
+	targets += V.owner.current
+	return targets
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/create_new_targeting()
+	var/datum/spell_targeting/select_vampire_network/T = new
+	return T
+
+/obj/effect/proc_holder/spell/vampire/thrall_commune/cast(list/targets, mob/user)
+	var/input = stripped_input(user, "Please choose a message to tell to the other thralls.", "Thrall Commune", "")
```

Comment:
I think this line sounds a little clumsy here. Maybe something like `"Enter a message to relay to the other thralls"`.